firefoxThemes/DokiThemeDefinitions.js

```diff
@@ -1 +1,2133 @@
+            "id": "63fe4617-4cac-47a5-9b93-6794514c35ad",
+            "name": "Megumin",
+            "displayName": "Megumin",
+            "dark": true,
+            "author": "Unthrottled",
+            "group": "KonoSuba",
+            "stickers": {"default": "megumin.png"},
+            "imagePath": "waifus/Megumin/images/megumin.png",
+            "jsonPath": "waifus/Megumin/theme.json"
+        },
+        "colors": {
+            "caretRow": "#31262b",
+            "caretForeground": "#ffffff",
+            "lineNumberColor": "#7b7373",
+            "infoForeground": "#75595d",
+            "completionWindowBackground": "#32393F",
+            "baseIconColor": "#948785",
+            "ghostColor": "#FFFFFF00",
+            "contrastColor": "#3e2126",
+            "nonProjectFileScopeColor": "#312A26",
+            "editorBackground": "#2f2329",
+            "secondaryBackground": "#2f2329",
+            "selectionForeground": "#E2CB14",
+            "inactiveBackground": "#2a272e",
+            "inactiveBackgroundDarker": "#333037",
+            "headerColor": "#2a1f24",
+            "baseBackground": "#271c21",
+            "borderColor": "#32262c",
+            "buttonColor": "#2f2329",
+            "selectionInactive": "#31262b",
+            "tagColor": "#7A7A7A",
+            "selectionBackground": "#563036",
+            "selectionBackgroundTransparent": "#563036A5",
+            "buttonFont": "#766B6B",
+            "foregroundColor": "#bbbbbb",
+            "startColor": "#511c22",
+            "highlightColor": "#a32f345a",
+            "disabledColor": "#585F6F",
+            "accentColorTransparent": "#a730355A",
+            "accentColorLessTransparent": "#a730359A",
+            "accentColorMoreTransparent": "#a730352A",
+            "accentColor": "#a73035",
+            "accentContrastColor": "#271c21",
+            "stopColor": "#ee2e37",
+            "testScopeColor": "#243125",
+            "popupMask": "#EAEAF00A",
+            "codeBlock": "#2f2329",
+            "textEditorBackground": "#2c2126",
+            "searchResultBackground": "#33282d",
+            "classNameColor": "#dae838",
+            "htmlTagColor": "#a28a92",
+            "stringColor": "#ffd866",
+            "keyColor": "#AF5C5A",
+            "keywordColor": "#a28a92",
+            "diff.deleted": "#3E3839"
+        },
+        "overrides": {
+            "theme": {
+                "colors": {
+                    "omnibox_text": "&selectionForeground&",
+                    "button_background_active": "#563136",
+                    "button_background_hover": "#563136",
+                    "tab_selected": "#563136",
+                    "tab_background_text": "&lineNumberColor&"
+                }
+            }
+        }
+    }, "dce48196-ff46-470c-b5f9-d1e23f4a79d3": {
+        "information": {
+            "id": "dce48196-ff46-470c-b5f9-d1e23f4a79d3",
+            "name": "Monika Dark Joy",
+            "displayName": "Monika (Joy)",
```

Comment:
1. Mkay I'll check out the contributing file. If I don't understand something I'll let you know.

2. I see. I added these initially because I liked the secondary background of Natsuki. But if there will be more secondary backgrounds coming in... That would be cool. If secondary backgrounds are supported, they can be placed into the popup menu under the category _Alternative_ or _Secondary_ or etc.

Reply:
>  If secondary backgrounds are supported, they can be placed into the popup menu under the category Alternative or Secondary or etc.

I don't know if adding more things to the dropdown selection menu is best usability option. The list is starting to get pretty long. Take for instance the hyper plugin.

![image](https://user-images.githubusercontent.com/15972415/103345885-2e579000-4a58-11eb-836e-53c777e9ff6b.png)

I think for the short term it could be added to the popup. Long term, if we could just give the user a list of just the distinct waifu, and allow them to set light/dark and which background they want (if supported)

Reply:
Well i have some solutions to this, maybe. 

### Popup Choice
One is to pretty much do as you stated here,
> Long term, if we could just give the user a list of just the distinct waifu, and allow them to set light/dark and which background they want (if supported)

But instead keep the dropdown menu idea. 

![mockup-1](https://user-images.githubusercontent.com/23222943/103369188-62b36680-4ac1-11eb-834b-ddb0be5fa70a.png)
The user can select the type of background to show from the left section of the popup menu. From the chosen category, the dropdown menu will be loaded with the themes in that category. So, if I choose dark, then the dropdown menu will be loaded with only dark themes.

Or the logo can be removed & the options takes its place like so:
![popup-2mockup](https://user-images.githubusercontent.com/23222943/103371437-5a5e2a00-4ac7-11eb-933d-0ae394277327.png)

### Sidebar Choice
The other option is a custom sidebar.
![sidebar-mock](https://user-images.githubusercontent.com/23222943/103369393-ea997080-4ac1-11eb-95ff-640dcded3bd6.png)
The user will first click on the doki theme icon on the top-right of the browser _as usual_, but instead of a popup menu, a sidebar will appear.

Now here's the interesting part. As you can see, the sidebar is pretty big (_disclaimer: this is the sidebar from Amazon_, which is only being used under fair use). It can pretty much host all of the Doki Themes currently available. As more themes are created, the custom sidebar can easily be adjusted to something like this.
![sidebar-mock-2](https://user-images.githubusercontent.com/23222943/103370564-05211900-4ac5-11eb-81c9-781fad81d8bf.png)

Additionally, this sidebar can appear on any [valid match](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Match_patterns#scheme) domain including `about:newtab`. To clarify, I can visit any site and if I want to change the theme the sidebar will always be there when needed.

Reply:
So I like the popup choice. 

Where it would have:

- A drop down of all the distinct waifu
- [A switch](https://material-ui.com/components/switches/#switch) to toggle light or dark (it would be disabled for a waifu that doesn't have both)
- Another switch to toggle the background between primary and secondary (it would be disabled for waifu that only have a single primary background)

However, that can be added some time after this request.

Reply:
> Another switch to toggle the background between primary and secondary (it would be disabled for waifu that only have a single primary background

:point_up: I went with this for the time being, I think that it might be nice to a light dark toggle as well (I'll make an issue for that).